source/extensions/http/original_ip_detection/xff/xff.cc

```diff
@@ -0,0 +1,29 @@
+#include "extensions/http/original_ip_detection/xff/xff.h"
+
+#include "common/http/utility.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Http {
+namespace OriginalIPDetection {
+namespace Xff {
+
+XffIPDetection::XffIPDetection(
+    const envoy::extensions::http::original_ip_detection::xff::v3::XffConfig& config)
+    : xff_num_trusted_hops_(config.xff_num_trusted_hops()) {}
+
+XffIPDetection::XffIPDetection(uint32_t xff_num_trusted_hops)
+    : xff_num_trusted_hops_(xff_num_trusted_hops) {}
+
+Envoy::Http::OriginalIPDetectionResult
+XffIPDetection::detect(Envoy::Http::OriginalIPDetectionParams& params) {
+  auto ret =
+      Envoy::Http::Utility::getLastAddressFromXFF(params.request_headers, xff_num_trusted_hops_);
+  return {ret.address_, ret.single_address_, absl::nullopt};
```

Comment:
Can you leave a comment trail here on why single_address_ means the address is trusted? This code is confusing and I just want to make sure people have a good comment trail to follow.

Reply:
I think instead of comment we should just rename that field here:

https://github.com/envoyproxy/envoy/blob/main/source/common/http/utility.h#L345

We are already doing it for ConnectionManagerUtility::mutateRequestHeaders():

```
-  bool single_xff_address;
+  bool allow_trusted_address_checks = false;
```

No point in keeping it around with the former name in other places... I'll update.